### ISSUE TITLE
registry login failed: 400

### ISSUE THREAD

**Issue by Junnplus** [containerd/nerdctl#717]
```
[N] ❯ nerdctl login registry.cn-shanghai.aliyuncs.com -u xxx -p xxx --debug
WARN[0000] WARNING! Using --password via the CLI is insecure. Use --password-stdin.
DEBU[0000] Ignoring hosts dir "/etc/containerd/certs.d"  error="stat /etc/containerd/certs.d: no such file or directory"
DEBU[0000] Ignoring hosts dir "/etc/docker/certs.d"      error="stat /etc/docker/certs.d: no such file or directory"
DEBU[0000] len(regHosts)=1
DEBU[0000] token request failed                          body="{\"error\":\"bad request\"}" status="400 Bad Request"
ERRO[0000] failed to call tryLoginWithRegHost            error="failed to call rh.Authorizer.Authorize: failed to fetch oauth token: unexpected status: 400 Bad Request" i=0
FATA[0000] failed to call rh.Authorizer.Authorize: failed to fetch oauth token: unexpected status: 400 Bad Request
```

**Comment by Junnplus** [comment]
Looks like an issue with the aliyun registry service, but `docker login` is working.

**Comment by Junnplus** [comment]
I checked the auth request parameters
```
access_type=offline&client_id=containerd-client&grant_type=password&password=xxx&scope=&service=registry.aliyuncs.com%3Acn-shanghai%3A26842&username=xxx
```
`scope=` results in a 400 error, remove it is working.

**Comment by chklauser** [comment]
> Looks like an issue with the aliyun registry service, but `docker login` is working.

I'm seeing the same issue with the Azure container registry (ACR). Login via docker / moby works.

**Comment by Junnplus** [comment]
@chklauser may be related to https://github.com/containerd/nerdctl/issues/715

**Comment by paulorpdlitss** [comment]
> > Looks like an issue with the aliyun registry service, but `docker login` is working.
> 
> I'm seeing the same issue with the Azure container registry (ACR). Login via docker / moby works.

I'm facing the same issue with ACR. Have cloned and built nerdctl on my env, after #718 merge, and it's still not working for me at least.

**Comment by Junnplus** [comment]
@paulorpdlitss @chklauser  This fix is about Aliyun container registry and it works, I'm not sure if it's the same problem with ACR, a new issue can be opened to discuss this issue.

**Comment by aellwein** [comment]
/reopen
This is still not working with ACR.

**Comment by AkihiroSuda** [comment]
@aellwein 

> /reopen
> This is still not working with ACR.

Alibaba’s ACR or Azure’s ACR?

**Comment by aellwein** [comment]
> Alibaba’s ACR or Azure’s ACR?
@AkihiroSuda not working on Azure's ACR, not tested on Alibaba

**Comment by fahedouch** [comment]
@aellwein which login option are using to login to ACR : 
1- username/password
2- 00000000-0000-0000-0000-000000000000/accessToken

can you share the error msg that you got plz

**Comment by aellwein** [comment]
@fahedouch @AkihiroSuda Ah sorry, my bad, i apparently forgot to take the -u 00000000-0000-0000-0000-000000000000 parameter. Now it works, "Login succeeded". Thanks!

Oneliner for those, who are using nerdctl on Azure CR (just replace ``myregistryname`` with your registry name):

```sh
az acr login -n myregistryname --expose-token --query 'accessToken' -o tsv | nerdctl login -u 00000000-0000-0000-0000-000000000000 --password-stdin myregistryname.azurecr.io
```
